Flushes the buffer at every message so the consumer can immediately them

hbina opened the pull request:
Without this, the user will only see the messages when the async thing times out. Which is not entirely useful IMO.

I think one issue is that if we are listening from the beginning, we will flush at every interval. I think what we can do is to have a buffer that can store a certain amount of bytes and flushes every messages (up to the message delimiter). But this poses an issue with messages that are larger than the buffer.

So I am not sure if there's any async tricks to stdout. I am new to this ^^

Signed-off-by: Hanif Bin Ariffin <hanif.ariffin.4326@gmail.com>

hbina committed:
Flushes the buffer at every message so the consumer can immediately see the messages

Signed-off-by: Hanif Bin Ariffin <hanif.ariffin.4326@gmail.com>

qiujiangkun commented:
I think we can have an option about flushing strategy. e.g. `-f` for consume mode, `-f 0` means immediate flush, `-f 100` means flush every 10 messages, `-f 100ms` means flush every 100ms or when buffer full.
Would you mind helping me on this?
btw there's a typo `does not supports {} ...`, please fix it

hbina committed:
Provides some ability to determine when flushing should be triggered.

Signed-off-by: Hanif Bin Ariffin <hanif.ariffin.4326@gmail.com>

hbina commented:
@qiujiangkun Hi, I have updated the PR to also adds 2 configuration for the flushing: bytes of messages read and the number of messages read. I have also updated the typo (I assume this is what you wanted).
